src/main/java/org/reaktivity/nukleus/oauth/internal/stream/OAuthProxyFactory.java

```diff
@@ -245,6 +260,58 @@ private RouteFW wrapRoute(
         return routeRO.wrap(buffer, index, index + length);
     }
 
+    private OAuthAccessGrant modifyGrant(
+        JsonWebSignature verified,
+        long affinity,
+        long connectAuthorization,
+        long expiresAtMillis)
+    {
+        String subject = null;
+        try
+        {
+            if(verified != null)
+            {
+                final JwtClaims claims = JwtClaims.parse(verified.getPayload());
+                subject = claims.getSubject();
+            }
+        }
+        catch (InvalidJwtException | JoseException | MalformedClaimException e)
+        {
+            // TODO: diagnostics?
+        }
```

Comment:
Suggest moving the above lines to `String resolveSubject(JsonWebSignature)` returning the `subject`.